src/controllers/clientController.js

```diff
@@ -137,3 +137,41 @@ exports.getChefs = async (req, res) => {
       .json({ error: `Error while retrieving chefs: ${error.message}` });
   }
 };
+
+exports.dishesRatings = async (req, res) => {
+  const token = req.cookies.jwt;
+  const decodedToken = decodeJwtToken(token);
+
+  const { id } = req.params;
+  const { rating } = req.body;
+
+  try {
+    const dish = await Dish.findById(id);
+
+    const existingRatingIndex = dish.ratings.findIndex((r) =>
+      r.userId.equals(decodedToken.userId)
+    );
+
+    if (existingRatingIndex !== -1) {
+      dish.ratings[existingRatingIndex].rating = rating;
+    } else {
+      const newRating = { userId: decodedToken.userId, rating };
+      dish.ratings.push(newRating);
+    }
+
+    const ratingsCount = dish.ratings.length;
+    const ratingSum = dish.ratings.reduce(
+      (sum, value) => sum + value.rating,
+      0
+    );
+    const averageRating = ratingsCount > 0 ? ratingSum / ratingsCount : 0;
```

Comment:
I think you don't need this ternary condition, cuz when the code reaches this point it definitely `ratingsCount > 0`

Reply:
done...